### PR TITLE
Add alpha tests (subsetting) to xds_k8s_lb

### PR DESCRIPTION
**Pull request by lidizheng**
This PR adds the ability to run xDS test against GCP Discovery alpha API. For non-stable features, we only need to test them with one language, so we are only adding them to the C++ test suite for now.

Test run: https://fusion2.corp.google.com/invocations/ef45392c-2d81-413a-9819-cdfd2e24c647

### CODE REVIEW

**Review by sergiitk — CHANGES_REQUESTED** [review]
If possible, I'd avoid saving alpha.json, so we don't have to keep it in sync. Can we access alpha api discovery doc similar to how we do it for networksecurity/networkservices?

**Comment by lidizheng** [comment]
@sergiitk PTALA. I have updated the logic to access the Compute alpha API. I considered letting test case decide which version to use, but the scope for each Compute API version is tricky for url-map tests.

**Comment by lidizheng** [comment]
Test ad-hoc Kokoro run passed.

File: tools/run_tests/xds_k8s_test_driver/framework/infrastructure/traffic_director.py [hunk 1]
```diff
@@ -68,9 +68,12 @@ def __init__(
         resource_prefix: str,
         resource_suffix: str,
         network: str = 'default',
+        compute_api_version: Optional[str] = None,
```
Comment (sergiitk) [inline tools/run_tests/xds_k8s_test_driver/framework/infrastructure/traffic_director.py:71]: Should be fine to have a default here too.

```suggestion
        compute_api_version: str = 'v1',
```

File: tools/run_tests/xds_k8s_test_driver/framework/infrastructure/gcp/compute.py [hunk 2]
```diff
@@ -40,8 +40,12 @@ class GcpResource:
     class ZonalGcpResource(GcpResource):
         zone: str
 
-    def __init__(self, api_manager: gcp.api.GcpApiManager, project: str):
-        super().__init__(api_manager.compute('v1'), project)
+    def __init__(self,
+                 api_manager: gcp.api.GcpApiManager,
+                 project: str,
+                 version: str = 'v1'):
+        # Version maybe also be v1alpha or v1beta
```
Comment (sergiitk) [inline tools/run_tests/xds_k8s_test_driver/framework/infrastructure/gcp/compute.py:47]: For now we don't support `v1beta` in GcpApiManager. I think we can skip this comment.
```suggestion
```

**Review by sergiitk — APPROVED** [review]
LGTM

**Comment by sergiitk** [comment]
Thank you so much for making the change with the alpha compute!

**Comment by dapengzhang0** [comment]
@lidizheng @sergiitk , this PR seems to cause PSM security tests failing. Can you PTAL? b/201535750

**Pull request merged**